cmake/project_version.cmake

```
@@ -0,0 +1,28 @@
+# Increment this appropriately at tag-and-release time.
+set(sail_riscv_release_version "0.7")
+
+# Sets GIT_EXECUTABLE
+find_package(Git)
+
+if (Git_FOUND)
+  # We could remove --tags if we started annotating the release tags.
+  execute_process(
+    COMMAND ${GIT_EXECUTABLE} describe --tags HEAD
```

The `--dirty` flag will indicate if there are local changes as well which is probably worth including.
```suggestion
    COMMAND ${GIT_EXECUTABLE} describe --dirty --tags HEAD
```